receiver/kubeletstatsreceiver/scraper.go

```diff
@@ -84,47 +88,42 @@ func (r *kubletScraper) scrape(context.Context) (pmetric.Metrics, error) {
 	}
 
 	metadata := kubelet.NewMetadata(r.extraMetadataLabels, podsMetadata, r.detailedPVCLabelsSetter())
-	mds := kubelet.MetricsData(r.logger, summary, metadata, r.metricGroupsToCollect)
+	mds := kubelet.MetricsData(r.logger, summary, metadata, r.metricGroupsToCollect, r.mb)
 	md := pmetric.NewMetrics()
 	for i := range mds {
 		mds[i].ResourceMetrics().MoveAndAppendTo(md.ResourceMetrics())
 	}
 	return md, nil
```

Comment:
This must be the only place where we call Emit, in other places we should use EmitForResource
```suggestion
	return r.mb.Emit(), nil
```

Reply:
What should MetricsData return, or should it return nothing?

Reply:
Ok I understand know.  Code updated